drivers/include/mtdi.h

```diff
@@ -0,0 +1,211 @@
+#define MTDI_GET_SECTOR_COUNT    3   /**< Get media size */
+#define MTDI_GET_SECTOR_SIZE     4   /**< Get sector size */
+#define MTDI_GET_BLOCK_SIZE      5   /**< Get erase block size */
+#define MTDI_CTRL_ERASE_BLOCK    6   /**< Force erased a block of blocks */
+#define MTDI_CTRL_POWER          7   /**< Get/Set power status */
+#define MTDI_CTRL_LOCK           8   /**< Lock/Unlock media removal */
+#define MTDI_CTRL_EJECT          9   /**< Eject media */
+/** @} */
+
+/**
+ * @name specific ioctl command
+ * @{
+ */
+#define MTDI_MMC_GET_TYPE        15  /**< Get memory type - NAND / NOR */
+#define MTDI_MMC_GET_SDSTAT      16  /**< Get memory status */
+/** @} */
+
+/**
+ * @name Memory type
+ * @{
+ */
+#define MEMORY_TYPE_NOR     20
+#define MEMORY_TYPE_NNAND   21
+/** @} */
+
+/** @} */
+
+typedef struct mtdi_desc mtdi_desc_t;
+
+typedef struct {
+    const mtdi_desc_t *driver;
+} mtdi_dev_t;
+
+
+/**
+ * @brief MTD driver interface
+ *
+ * This define the functions to access a MTD.
+ *
+ * A MTD is composed of pages combined into sectors. A sector is the smallest erasable unit.
+ * The number of pages in a sector must be constant for the whole MTD.
+ *
+ * Most of the flash filesystems implementation will use pages as basic unit for writing.
+ * This interface does not limit the writing on bage boundaries but user must not write on
+ * multiple sector.
+ *
+ * The erase is available only for entire sectors.
+ */
+struct mtdi_desc {
+    /**
+     * @brief Initialize Memory Technology Device (MTD)
+     *
+     * @param[in] mtdi_dev  Pointer to the selected driver
+     *
+     * @returns 0 on success
+     * @returns any other @ref mtdi_sta_t value on error
+     */
+    mtdi_sta_t (*mtdi_init)(mtdi_dev_t *mtdi_dev);
+
+
+    /**
+     * @brief Get the status of the Memory Technology Device (MTD)
+     *
+     * @param[in] mtdi_dev  Pointer to the selected driver
+     *
+     * @returns 0 on success
+     * @returns any other @ref mtdi_sta_t value on error
+     */
+    mtdi_sta_t (*mtdi_status)(mtdi_dev_t *mtdi_dev);
+
+
+    /**
+     * @brief Read from the Memory Technology Device (MTD)
+     *
+     * No alignment is required on @p addr and @p size.
+     *
+     * @param[in] mtdi_dev  Pointer to the selected driver
+     * @param[out] buff     Pointer to the data buffer to store read data
+     * @param[in]  addr     Starting address
+     * @param[in]  size     Number of bytes
+     *
+     * @return @ref MTDI_RES_OK on success
+     * @return any other @ref mtdi_result_t value on error
+     */
+    mtdi_result_t (*mtdi_read)(mtdi_dev_t *mtdi_dev,
+                               unsigned char *buff,
+                               unsigned long addr,
+                               unsigned long size);
+
+
+    /**
+     * @brief Write to the Memory Technology Device (MTD)
+     *
+     * @p addr + @p size must be inside a sector boundary. @p addr can be anywhere
+     * but the buffer cannot overlap two sectors.
```

Comment:
I didn't seen any big issue with it, but I'm not sure if it really makes sense, so I will change the comment to allow only one page at a time